apis/v1alpha2/shared_types.go

```diff
@@ -20,63 +20,96 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// GatewayAllowType specifies which Gateways should be allowed to use a Route.
-type GatewayAllowType string
-
-const (
-	// Any Gateway will be able to use this route.
-	GatewayAllowAll GatewayAllowType = "All"
-	// Only Gateways that have been  specified in GatewayRefs will be able to use this route.
-	GatewayAllowFromList GatewayAllowType = "FromList"
-	// Only Gateways within the same namespace as the route will be able to use this route.
-	GatewayAllowSameNamespace GatewayAllowType = "SameNamespace"
-)
+// ParentRef identifies an API object (usually a Gateway) that can be considered
+// a parent of this resource (usually a route). The only kind of parent resource
+// with "Core" support is Gateway. This API may be extended in the future to
+// support additional kinds of parent resources, such as HTTPRoute.
+type ParentRef struct {
+	// Group is the group of the referent.
+	//
+	// Support: Core
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:default=gateway.networking.k8s.io
+	// +optional
+	Group *string `json:"group,omitempty"`
 
-// RouteGateways defines which Gateways will be able to use a route. If this
-// field results in preventing the selection of a Route by a Gateway, an
-// "Admitted" condition with a status of false must be set for the Gateway on
-// that Route.
-type RouteGateways struct {
-	// Allow indicates which Gateways will be allowed to use this route.
-	// Possible values are:
-	// * All: Gateways in any namespace can use this route.
-	// * FromList: Only Gateways specified in GatewayRefs may use this route.
-	// * SameNamespace: Only Gateways in the same namespace may use this route.
+	// Kind is kind of the referent.
+	//
+	// Support: Core (Gateway)
+	// Support: Extended (Other Resources)
 	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:default=Gateway
 	// +optional
-	// +kubebuilder:validation:Enum=All;FromList;SameNamespace
-	// +kubebuilder:default=SameNamespace
-	Allow *GatewayAllowType `json:"allow,omitempty"`
+	Kind *string `json:"kind,omitempty"`
 
-	// GatewayRefs must be specified when Allow is set to "FromList". In that
-	// case, only Gateways referenced in this list will be allowed to use this
-	// route. This field is ignored for other values of "Allow".
+	// Namespace is the namespace of the referent. When unspecified (empty
+	// string), this will either be:
```

Comment:
```suggestion
	// Namespace is the namespace of the referent. When unspecified (or empty
	// string), this will either be:
```

Namespace is a pointer.